lib/gemstash/upstream.rb

```diff
@@ -11,10 +11,11 @@ class Upstream
 
     attr_reader :user_agent, :uri
 
-    def_delegators :@uri, :scheme, :host, :user, :password, :to_s
+    def_delegators :@uri, :scheme, :host, :to_s
 
     def initialize(upstream, user_agent: nil)
-      @uri = URI(CGI.unescape(upstream.to_s))
+      url = upstream.to_s.start_with?("http") ? upstream.to_s : "https://#{upstream}"
```

Comment:
Q: Would it be prudent to do `.start_with("http://")` or `.start_with("https://")` so that inputs that are bare domain names that begin with http, such as `httpstatus.es` etc, are covered?